js/component.js

```diff
@@ -0,0 +1,25 @@
+/** @type {HTMLCanvasElement} */
+
+class Component {
+    constructor(x, y, w, h, color, ctx) {
+        this.x = x;
+        this.y = y;
+        this.w = w;
+        this.h = h;
+        this.color = color;
+        this.ctx = ctx;
+        this.speedX = 0;
+    }
+
+    draw() {
+        this.ctx.fillStyle = this.color;
+        this.ctx.fillRect(this.x, this.y, this.w, this.h);
+    }
+
+    newPosition() {
+        this.x += this.speedX;
+    }
+
+    // methods for the square of the enemy
+
```

Comment:
Here is where you could add your player class. 
Here's one example for you to add the image:
```
class Player {
  constructor(x, y, w, h, ctx) {
    this.x = x;
    this.y = y;
    this.w = w;
    this.h = h;
    this.ctx = ctx;
    this.car = new Image();
  }

  // Drawing the car (player)
  drawCar() {
    this.car.src = '/images/car.png';
    ctx.drawImage(this.car, this.x, this.y, this.w, this.h);
  }
```